src/telliot_feeds/sources/mimicry/mimicry_nft_market_index.py

```diff
@@ -0,0 +1,85 @@
+import asyncio
+import time
+from dataclasses import dataclass
+from typing import Any
+from typing import Optional
+
+from requests import JSONDecodeError
+from requests.adapters import HTTPAdapter
+from requests.adapters import Retry
+from requests.exceptions import ConnectionError
+from requests.exceptions import HTTPError
+from requests.exceptions import RequestException
+from requests.exceptions import Timeout
+from requests.sessions import Session
+from telliot_core.apps.telliot_config import TelliotConfig
+
+from telliot_feeds.datasource import DataSource
+from telliot_feeds.dtypes.datapoint import datetime_now_utc
+from telliot_feeds.dtypes.datapoint import OptionalDataPoint
+from telliot_feeds.utils.log import get_logger
+
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class NFTGoSource(DataSource[str]):
+    """DataSource for NFTGo expected response data."""
+
+    metric_currency: Optional[str] = None
+    api_key = TelliotConfig().api_keys.find(name="nftgo")[0].key
+
+    async def fetch_nftgo_api(self, retries: int = 3) -> Optional[Any]:
+        """
+        Request NFTGo data from the api
+        """
+
+        url = "https://data-api.nftgo.io/eth/v1/market/rank/collection/all"
+        headers = {"X-API-KEY": self.api_key, "accept": "application/json"}
+        params = {"by": "market_cap", "with_rarity": "false", "asc": "false", "offset": 0, "limit": 50}
+
+        with Session() as session:
+            session.headers.update(headers)
+            retry_strat = Retry(total=retries, backoff_factor=0.5, status_forcelist=[500, 502, 503, 504])
+            adapter = HTTPAdapter(max_retries=retry_strat)
```

Comment:
Retry strat and adapter can be instantiate once above the source class instead of each time the function is called